### PR TITLE
ENG-8685: Add Duo MFA integration resource

### PR DESCRIPTION
**Pull request by Yowgf**
## Description of the change

This PR adds the resource `cyral_integration_mfa_duo`, and refactors the code for resource `cyral_integration_pager_duty`. The API calls these two resources make are largely the same, so I decided to reuse the code. The model can be found in `model_integration_confextension.go`.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Jira issue referenced in commit message and/or PR title

### Testing

- Added acceptance tests for Duo MFA resource
- Acceptance tests for Pager Duty are passing
- Tested manually both Duo MFA and Pager Duty resources and checked that integration was created in the control plane. Deletion was also successful for both.

### CODE REVIEW

**Comment by Yowgf** [comment]
@VictorGFM just FYI I saw that in the `cyral_integration_pager_duty` resource we allow empty values for the arguments `name` and `api_token`. However, if the user leaves these parameters empty, it breaks the control plane UI. I didn't add the `validation.StringIsNotEmpty` validation to avoid backward incompatibility issues.

File: cyral/resource_cyral_integration_mfa_duo.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+				Sensitive:    true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"api_hostname": {
+				Description:  "API hostname obtained from Duo management console.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
```
Comment (VictorGFM) [inline cyral/resource_cyral_integration_mfa_duo.go:51]: Im assuming that all these fields are in fact required by the API.

**Review by VictorGFM — APPROVED** [review]
LGTM @Yowgf, nice refactoring 🙂

**Comment by VictorGFM** [comment]
> @VictorGFM just FYI I saw that in the `cyral_integration_pager_duty` resource we allow empty values for the arguments `name` and `api_token`. However, if the user leaves these parameters empty, it breaks the control plane UI. I didn't add the `validation.StringIsNotEmpty` validation to avoid backward incompatibility issues.

How does the UI break? I tried to reproduce that on `stable.dev` by creating a pager duty integration with empty fields but the UI seem to work just fine after the integration was created. If this is actually breaking the UI lets file a ticket and post on the Jeeves channel so that this can be tracked and fixed in the near future.

File: cyral/resource_cyral_integration_mfa_duo.go (hunk 1 above)
Comment (Yowgf) [reply cyral/resource_cyral_integration_mfa_duo.go:51]: Yep! If you don's specify them, the API returns an error.

**Comment by VictorGFM** [comment]
Both `cyral_integration_pager_duty` and `cyral_integration_mfa_duo` were tested and worked just fine without impacting the UI (even when empty fields were specified for the resources). That being said, this PR is ready to be merged.

**Pull request merged**